yellowbrick/base.py

```diff
@@ -407,3 +408,119 @@ def predict(self, X, y):
+        Specify only one of nrows and ncols, the other should be None. If you 
+        specify nrows, there will be enough columns created to fit all the 
+        visualizers specified in the visualizers list. 
+    
+    ncols: integer, default: None
+        The number of columns desired, if you would like a fixed number of columns.
+        Specify only one of nrows and ncols, the other should be None. If you 
+        specify ncols, there will be enough rows created to fit all the 
+        visualizers specified in the visualizers list. 
+
+    axarr: matplotlib.axarr, default: None.
+        If you want to put the plot onto an existing axarr, specify it here. Otherwise a new
+        one will be created. 
+
+    kwargs : additional keyword arguments, default: None
+        Any additional keyword arguments will be passed on to the fit() method and therefore 
+        passed on to the fit() method of the wrapped estimators, if applicable. Otherwise ignored.
+
+    Examples
+    --------
+    >>> from yellowbrick.base import MultipleVisualizer
+    >>> from sklearn.linear_model import LogisticRegression
+    >>> from yellowbrick.classifier import ConfusionMatrix
+    >>> from yellowbrick.classifier import ClassBalance
+    >>> model = LogisticRegression()
+    >>> visualizers = [ClassBalance(model),ConfusionMatrix(model)]
+    >>> mv = MultipleVisualizer(visualizers, ncols=2)
+    >>> mv.fit(X_train, y_train)
+    >>> mv.score(X_test, y_test)
+    >>> mv.poof()
+    """
+    def __init__(self, visualizers = [], nrows = None, ncols = None, axarr = None, **kwargs):
+        self.visualizers = visualizers
+        self.plotcount = len(visualizers)
+        if nrows == None and ncols == None:
+            self.ncols = 1
+            self.nrows = self.plotcount
+        elif ncols == None:
+            self.nrows = nrows
+            self.ncols = math.ceil(self.plotcount / self.nrows)
+        elif nrows == None:
+            self.ncols = ncols
+            self.nrows = math.ceil(self.plotcount / self.ncols)
+        else:
+            raise YellowbrickValueError("You can only specify either nrows or ncols, \
+                the other will be calculated based on the length of the list of visualizers.")
+        
+
+        if axarr == None:
+            fig, axarr = plt.subplots(self.nrows, self.ncols, squeeze = False)
+        
+        self.axarr = axarr
+
+        idx = 0
+        for row in range(self.nrows):
+            for col in range(self.ncols):
+                try:
+                    self.visualizers[idx].ax = self.axarr[row, col]
+                #If len(visualizers) isn't evenly divisibly by rows/columns, 
+                #we want to create the illusion of empty space by hiding the axis
+                except IndexError as e:
+                    self.axarr[row,col].axis('off')
+
+                idx += 1
+
+        self.kwargs = kwargs
+
+    def fit(self,X,y):
+
+        for idx in range(len(self.visualizers)):
```

Comment:
Just out of curiosity, is there a reason not to just loop over `self.visualizers`?

Reply:
I think it's that when I first wrote this a year ago I wasn't 100% certain that a copy of the visualizer would be created with `for viz in self.visualizers` - this one is clearly operating on the element in place. Rather than double check I just wrote it that way. I can switch to `self.visualizers`

Reply:
changed